### PR TITLE
Modernizing the codebase - checking for nullptr

### PR DESCRIPTION
**Pull request by JanVogelsang**
**This PR is based on PR #2493.**  
In the C++ standard it is defined that `if ( ptr != nullptr )` gives the same result as `if ( ptr )` as `ptr` is converted to a `bool` in the latter case, which per standard evaluates to `true` only if `ptr != nullptr`.  
We can therefore convert all occurrences of
`if ( ptr != nullptr )` to ` if ( ptr )`
and
`if ( ptr == nullptr )` to `if ( not ptr )`
which makes it quite clear in which cases a pointer is expected to be valid and when not.

### CODE REVIEW

File: models/aeif_psc_delta.cpp [hunk 1]
```diff
@@ -352,15 +352,15 @@ nest::aeif_psc_delta::init_buffers_()
...
+  if ( B_.s_ == nullptr )
   {
     B_.s_ = gsl_odeiv_step_alloc( gsl_odeiv_step_rkf45, State_::STATE_VEC_SIZE );
   }
   else
   {
     gsl_odeiv_step_reset( B_.s_ );
   }
-  if ( B_.c_ == 0 )
+  if ( B_.c_ == nullptr )
```
Comment (jougs) [inline models/aeif_psc_delta.cpp:363]: Same here

File: models/aeif_psc_delta.cpp [hunk 2]
```diff
@@ -369,7 +369,7 @@ nest::aeif_psc_delta::init_buffers_()
     gsl_odeiv_control_init( B_.c_, P_.gsl_error_tol, P_.gsl_error_tol, 0.0, 1.0 );
   }
 
-  if ( B_.e_ == 0 )
+  if ( B_.e_ == nullptr )
```
Comment (jougs) [inline models/aeif_psc_delta.cpp:372]: And here and probably more below...

**Comment by JanVogelsang** [comment]
Not sure why, but after the latest commit, I now double-checked both `== nullptr` and `!= nullptr` and did not find any more occurrences.

**Review by jougs — APPROVED** [review]
What can I say?!

**Review by heplesser — APPROVED** [review]
@JanVogelsang Looks good, thank you! In the next PR (replacing `&&`, `||`, and `!` with `and`, `or` and `not`), it would be good to also remove superfluous parentheses in boolean expression, e.g. of to form `(n1) and (n2)`.

**Comment by heplesser** [comment]
And merging ...

**Comment by JanVogelsang** [comment]
> @JanVogelsang Looks good, thank you! In the next PR (replacing `&&`, `||`, and `!` with `and`, `or` and `not`), it would be good to also remove superfluous parentheses in boolean expression, e.g. of to form `(n1) and (n2)`.

I could only find very few occurrences (~25) of redundant parentheses, but I think I got them all. PR is incoming as soon as all tests pass.

**Pull request merged**